### ISSUE TITLE
Antialiasing and adjacent shapes

### ISSUE THREAD

**Issue by lucach** [LuCEresearchlab/pytamaro#34]
With cb78384e0c15b6b2298162f62c0112f36abb74db antialiasing is enabled by default, which is great for many of our graphics.

Depending on the specific size and combinators used, some shapes might not line up well. Example:

```py
show_graphic(compose(
        pin(bottom_center, triangle(200, 200, 60, white)),
        pin(bottom_center, triangle(220, 220, 60, red))
))
```
![image](https://github.com/LuCEresearchlab/pytamaro/assets/2065805/4d5a6cfd-0ac0-4584-94b5-5ad850e546c4)

This has been discussed on Skia's forum [here](https://groups.google.com/g/skia-discuss/c/mU4X9An_844/m/Zk-l2nXxAwAJ) and [here](https://groups.google.com/g/skia-discuss/c/NlKpD2X_5uc/m/Vuwd-kyYBwAJ).

> This is working as intended. For normal AA each shape is drawn independently. So after one shape is draw, pixels around the edge will have some level of anti aliasing applied to them. In other words they will look lighter than the main paint to take into account the partial coverage. However, once we draw a pixel we no longer know what part of a pixel was partially covered. All we see is that there is a pixel with a lighter color. So then come the second draw, it is simply blending as if the whole destination was that lighter color. For non-AA this doesn't happen cause a pixel that sits along a shared edge will only ever belong to one shape or the other. So there is no double blending.
>
> The only type of AA that allows shapes drawn next to each other to seam correctly is using multisampling (i.e. MSAA). This is because it basically acts like the non-AA case and individual sample locations with a pixel will only belong to what shape of the other along the edge.

Assuming we don't use AA features from the GPU (e.g., MSAA), one of the workaround proposed (rendering the fill without AA, and separately the stroke with AA) works well for this case:

![image](https://github.com/LuCEresearchlab/pytamaro/assets/2065805/52b72446-fc3b-4e21-adfb-baeca2f5cefc)

but doesn't work with semi-transparent fills:

![image](https://github.com/LuCEresearchlab/pytamaro/assets/2065805/5d35d61e-bd3f-4711-9353-f5a57309b18d)

We could adopt the workaround, which might be good enough for our purposes, but I'd first like to collect here some example graphics to better understand the extent of this problem. Reply here with the code when you see one in the wild.

**Comment by frovaaa** [comment]
One side-effect of the antialiasing is that now internal lines of complex shapes are more visible

Nested flower activity:
https://pytamaro.si.usi.ch/activities/luce/nested-flower/en/v1

NO Antialiasing:

![image](https://github.com/user-attachments/assets/304e2743-4892-4c8a-83ac-37eb1301d4c5)


With Antialiasing:

![image](https://github.com/user-attachments/assets/89ea7fc4-34c4-457e-8aca-212fa6591ad4)



Heart activity:
https://pytamaro.si.usi.ch/activities/luce/heart/en/v2

NO Antialiasing:

![image](https://github.com/user-attachments/assets/9169c45d-6eac-4b98-8584-4425e3cd98d1)



With Antialiasing:

![image](https://github.com/user-attachments/assets/e481d6c7-10d9-4052-9ea5-3c9edb8b9824)

**Comment by lucach** [comment]
Workaround (fill without AA, hairline stroke with AA) implemented in [52c07c7](https://github.com/LuCEresearchlab/pytamaro/commit/52c07c77c57787b23df594177eb5bf4c0b534804).

If someone still bumps into seams, let's post them here so that we can investigate.

**Comment by lucach** [comment]
Commit https://github.com/LuCEresearchlab/pytamaro/commit/4244cc44b4011aadaa6ff7fbb42b0f0265e516fc implements 4x SSAA (supersampling) to get antialiasing. This avoids all the issues reported above (seams) with Skia's analytical antialiasing, as well as other issues with patches (problems with circular shapes at the boundaries due to stroking). The performance cost seems acceptable -- we have been testing it for a week.

The fix is part of the newly released v0.7.3.
